### PR TITLE
BRS-971-4: update KC roles on add/delete legacy item

### PR DESCRIPTION
**Pull request by cameronpettit**
### Jira Ticket:
BRS-971

### Jira Ticket URL:
https://bcparksdigital.atlassian.net/browse/BRS-971

### Description:
Last change necessary for the migration - this change is to include the addition/deletion of KC roles when migrating/deleting legacy records. To run migrate now you must provide the target env and a valid token for the env to run: 

```
node migrate.js dev <token>
```

or 

```
node purgeLegacy.js dev <token>
```

### CODE REVIEW

File: migrations-data/legacy-data/2017-2019/migrate.js [hunk 1]
```diff
@@ -362,10 +391,30 @@ async function executeChangelog() {
       let intervalStartTime = new Date().getTime();
       for (const transaction of subAreaTransactObjList) {
         updateConsoleProgress(intervalStartTime, 'Executing legacy subarea transaction', subAreaTransactObjList.indexOf(transaction) + 1, subAreaTransactObjList.length, 1);
+        for (const item of transaction.TransactItems) {
+          try {
+            // create KC roles
+            let key = AWS.DynamoDB.Converter.unmarshall(item?.Put?.Item);
```
Comment (marklise) [inline migrations-data/legacy-data/2017-2019/migrate.js:397]: `const key`

File: migrations-data/legacy-data/2017-2019/migrate.js [hunk 2]
```diff
@@ -362,10 +391,30 @@ async function executeChangelog() {
...
+            if (key.orcs && key.sk && key.parkName && key.subAreaName) {
+              let role = {
```
Comment (marklise) [inline migrations-data/legacy-data/2017-2019/migrate.js:399]: `const role`

**Review by marklise — APPROVED** [review]

File: migrations-data/legacy-data/2017-2019/migrate.js (hunk 1 above)
Comment (cameronpettit) [reply migrations-data/legacy-data/2017-2019/migrate.js]: resolved in `3e6c59`

File: migrations-data/legacy-data/2017-2019/migrate.js (hunk 2 above)
Comment (cameronpettit) [reply migrations-data/legacy-data/2017-2019/migrate.js]: resolved in `3e6c59`

**Review by danieltruong — APPROVED** [review]

**Pull request merged**